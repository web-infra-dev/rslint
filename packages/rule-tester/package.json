{
  "name": "@typescript-eslint/rule-tester",
  "version": "0.1.4",
  "type": "module",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "private": true,
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "import": "./dist/index.js",
      "default": "./dist/"
    },
    "./package.json": "./package.json"
  },
  "scripts": {
    "build": "tsc -b tsconfig.build.json --force",
    "dev": "tsc -b tsconfig.build.json --force --watch",
    "test": "rstest run --testTimeout=0",
    "test:update": "rstest run -u --testTimeout=0"
  },
  "files": [
    "dist"
  ],
  "keywords": [
    "rslint",
    "rule-tester",
    "testing",
    "linter"
  ],
  "author": "",
  "license": "MIT",
  "description": "Rule testing utilities for rslint",
  "devDependencies": {
    "@types/node": "24.0.14",
    "typescript": "5.8.3"
  },
  "dependencies": {
    "@rslint/core": "workspace:*"
  },
  "publishConfig": {
    "access": "public"
  }
}
