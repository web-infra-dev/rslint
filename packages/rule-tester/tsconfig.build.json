{
  "extends": "../../tsconfig.build.json",
  "compilerOptions": {
    "outDir": "dist",
    "rootDir": "src",
    "noEmit": false
  },
  "include": ["src/**/*"],
  "exclude": ["dist", "node_modules"]
}
