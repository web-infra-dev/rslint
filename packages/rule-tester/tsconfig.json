{
  "extends": "../../tsconfig.spec.json",
  "include": ["src/**/*", "tests/**/*"],
  "exclude": ["dist", "node_modules"]
}
