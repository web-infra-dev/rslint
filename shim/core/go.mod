module github.com/microsoft/typescript-go/shim/core

go 1.24.2

toolchain go1.24.4

require (
	github.com/microsoft/typescript-go v0.0.0-20250812073229-cedc0cbe6c18
	golang.org/x/text v0.27.0
)

require github.com/go-json-experiment/json v0.0.0-20250725192818-e39067aee2d2 // indirect
