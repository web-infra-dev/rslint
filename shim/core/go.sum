github.com/go-json-experiment/json v0.0.0-20250714165856-be8212f5270d h1:+d6m5Bjvv0/RJct1VcOw2P5bvBOGjENmxORJYnSYDow=
github.com/go-json-experiment/json v0.0.0-20250714165856-be8212f5270d/go.mod h1:TiCD2a1pcmjd7YnhGH0f/zKNcCD06B029pHhzV23c2M=
github.com/google/go-cmp v0.7.0 h1:wk8382ETsv4JYUZwIsn6YpYiWiBsYLSJiTsyBybVuN8=
github.com/google/go-cmp v0.7.0/go.mod h1:pXiqmnSA92OHEEa9HXL2W4E7lf9JzCmGVUdgjX3N/iU=
github.com/microsoft/typescript-go v0.0.0-20250725221625-c05da65ec429 h1:2GZhN4S7dal2oDXC8EOq7JGR5M1rLMLGfNGHj86fMw8=
github.com/microsoft/typescript-go v0.0.0-20250725221625-c05da65ec429/go.mod h1:ePDznrxm94dsru3ugv7f/F7INl3PSrG9U09gfw/Yu5Q=
gotest.tools/v3 v3.5.2 h1:7koQfIKdy+I8UTetycgUqXWSDwpgv193Ka+qRsmBY8Q=
gotest.tools/v3 v3.5.2/go.mod h1:LtdLGcnqToBH83WByAAi/wiwSFCArdFIUV/xxN4pcjA=
