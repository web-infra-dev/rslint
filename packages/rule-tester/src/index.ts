// Forked and modified from https://github.com/typescript-eslint/typescript-eslint/blob/16c344ec7d274ea542157e0f19682dd1930ab838/packages/rule-tester/src/RuleTester.ts#L4

import path from 'node:path';
import { test, describe, expect } from '@rstest/core';
import { lint, type Diagnostic } from '@rslint/core';
import assert from 'node:assert';

interface TsDiagnostic {
  line: number;
  column: number;
  endLine: number;
  endColumn: number;
  messageId: string;
  suggestions: any[];
}
function toCamelCase(name: string): string {
  return name.replace(/-([a-z])/g, g => g[1].toUpperCase());
}
// check whether rslint diagnostics and typescript-eslint diagnostics are semantic equal
function checkDiagnosticEqual(
  rslintDiagnostic: Diagnostic[],
  tsDiagnostic: TsDiagnostic[],
) {
  assert(
    rslintDiagnostic.length === tsDiagnostic.length,
    `Length mismatch: ${rslintDiagnostic.length} !== ${tsDiagnostic.length}`,
  );
  for (let i = 0; i < rslintDiagnostic.length; i++) {
    const rslintDiag = rslintDiagnostic[i];
    const tsDiag = tsDiagnostic[i];
    // check rule match
    assert(
      toCamelCase(rslintDiag.messageId) === tsDiag.messageId,
      `Message mismatch: ${rslintDiag.messageId} !== ${tsDiag.messageId}`,
    );

    // check range match
    // tsDiag sometimes doesn't have line and column, so we need to check that
    if (tsDiag.line) {
      assert(
        rslintDiag.range.start.line === tsDiag.line,
        `Start line mismatch: ${rslintDiag.range.start.line} !== ${tsDiag.line}`,
      );
    }
    if (tsDiag.endLine) {
      assert(
        rslintDiag.range.end.line === tsDiag.endLine,
        `End line mismatch: ${rslintDiag.range.end.line} !== ${tsDiag.endLine}`,
      );
    }
    if (tsDiag.column) {
      assert(
        rslintDiag.range.start.column === tsDiag.column,
        `Start column mismatch: ${rslintDiag.range.start.column} !== ${tsDiag.column}`,
      );
    }
    if (tsDiag.endColumn) {
      assert(
        rslintDiag.range.end.column === tsDiag.endColumn,
        `End column mismatch: ${rslintDiag.range.end.column} !== ${tsDiag.endColumn}`,
      );
    }
  }
}

interface RuleTesterOptions {
  languageOptions?: {
    parserOptions?: {
      project?: string;
      tsconfigRootDir?: string;
    };
  };
}
export class RuleTester {
  options: RuleTesterOptions;
  constructor(options: RuleTesterOptions) {
    this.options = options;
  }
  public run(
    ruleName: string,
    cases: {
      valid: (
        | string
        | { code: string; options?: any; only?: boolean; skip?: boolean }
      )[];
      invalid: {
        code: string;
        errors: any[];
        options?: any;
        only?: boolean;
        skip?: boolean;
      }[];
    },
  ) {
    describe(ruleName, () => {
      let cwd =
        this.options.languageOptions?.parserOptions?.tsconfigRootDir ||
        process.cwd();
      const config = path.resolve(cwd, './rslint.json');
      let virtual_entry = path.resolve(cwd, 'src/virtual.ts');
      // test whether case has only
      let hasOnly =
        cases.valid.some(x => {
          if (typeof x === 'object' && x.only) {
            return true;
          } else {
            return false;
          }
        }) || cases.invalid.some(x => x.only);
      test('valid', async () => {
        for (const validCase of cases.valid) {
          if (typeof validCase === 'object' && validCase.skip) {
            continue;
          }
          if (hasOnly) {
            if (typeof validCase === 'string') {
              continue;
            }
            if (!validCase.only) {
              continue;
            }
          }
          const code =
            typeof validCase === 'string' ? validCase : validCase.code;

          const options =
            typeof validCase === 'string' ? [] : validCase.options || [];

          // workaround for this hardcoded path https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/tests/rules/no-floating-promises.test.ts#L712
          if (Array.isArray(options)) {
            for (const opt of options) {
              if (Array.isArray(opt.allowForKnownSafeCalls)) {
                for (const item of opt.allowForKnownSafeCalls) {
                  if (item.path) {
                    item.path = virtual_entry;
                  }
                }
              }
            }
          }
          const diags = await lint({
            config,
            workingDirectory: cwd,
            fileContents: {
              [virtual_entry]: code,
            },
            ruleOptions: {
              [ruleName]: options,
            },
          });

          assert(
            diags.diagnostics?.length === 0,
            `Expected no diagnostics for valid case, but got: ${JSON.stringify(diags)}`,
          );
        }
      });
      test('invalid', async t => {
        for (const item of cases.invalid) {
          const {
            code,
            errors,
            only = false,
            skip = false,
            options = [],
          } = item;
          if (skip) {
            continue;
          }
          if (hasOnly && !only) {
            continue;
          }
          const diags = await lint({
            config,
            workingDirectory: cwd,
            fileContents: {
              [virtual_entry]: code,
            },
            ruleOptions: {
              [ruleName]: options,
            },
          });
          expect(diags).toMatchSnapshot();

          assert(
            diags.diagnostics?.length > 0,
            `Expected diagnostics for invalid case`,
          );
          checkDiagnosticEqual(diags.diagnostics, errors);
        }
      });
    });
  }
}

/**
 * Simple no-op tag to mark code samples as "should not format with prettier"
 *   for the plugin-test-formatting lint rule
 */
export function noFormat(raw: TemplateStringsArray, ...keys: string[]): string {
  return String.raw({ raw }, ...keys);
}
