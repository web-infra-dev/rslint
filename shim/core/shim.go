
// Code generated by tools/gen_shims. DO NOT EDIT.

package core

import "context"
import "github.com/microsoft/typescript-go/internal/core"
import "golang.org/x/text/language"
import "iter"
import _ "unsafe"

//go:linkname BoolToTristate github.com/microsoft/typescript-go/internal/core.BoolToTristate
func BoolToTristate(b bool) core.Tristate
type CompilerOptions = core.CompilerOptions
//go:linkname ComputeLineStarts github.com/microsoft/typescript-go/internal/core.ComputeLineStarts
func ComputeLineStarts(text string) []core.TextPos
//go:linkname ComputeLineStartsSeq github.com/microsoft/typescript-go/internal/core.ComputeLineStartsSeq
func ComputeLineStartsSeq(text string) iter.Seq[core.TextPos]
var ExclusivelyPrefixedNodeCoreModules = core.ExclusivelyPrefixedNodeCoreModules
//go:linkname GetLocale github.com/microsoft/typescript-go/internal/core.GetLocale
func GetLocale(ctx context.Context) language.Tag
//go:linkname GetRequestID github.com/microsoft/typescript-go/internal/core.GetRequestID
func GetRequestID(ctx context.Context) string
//go:linkname GetScriptKindFromFileName github.com/microsoft/typescript-go/internal/core.GetScriptKindFromFileName
func GetScriptKindFromFileName(fileName string) core.ScriptKind
//go:linkname IndexAfter github.com/microsoft/typescript-go/internal/core.IndexAfter
func IndexAfter(s string, pattern string, startIndex int) int
type JsxEmit = core.JsxEmit
const JsxEmitNone = core.JsxEmitNone
const JsxEmitPreserve = core.JsxEmitPreserve
const JsxEmitReact = core.JsxEmitReact
const JsxEmitReactJSX = core.JsxEmitReactJSX
const JsxEmitReactJSXDev = core.JsxEmitReactJSXDev
const JsxEmitReactNative = core.JsxEmitReactNative
type LanguageVariant = core.LanguageVariant
const LanguageVariantJSX = core.LanguageVariantJSX
const LanguageVariantStandard = core.LanguageVariantStandard
type LinkStore[K comparable, V any] = core.LinkStore[K,V]
type ModuleDetectionKind = core.ModuleDetectionKind
const ModuleDetectionKindAuto = core.ModuleDetectionKindAuto
const ModuleDetectionKindForce = core.ModuleDetectionKindForce
const ModuleDetectionKindLegacy = core.ModuleDetectionKindLegacy
const ModuleDetectionKindNone = core.ModuleDetectionKindNone
type ModuleKind = core.ModuleKind
const ModuleKindAMD = core.ModuleKindAMD
const ModuleKindCommonJS = core.ModuleKindCommonJS
const ModuleKindES2015 = core.ModuleKindES2015
const ModuleKindES2020 = core.ModuleKindES2020
const ModuleKindES2022 = core.ModuleKindES2022
const ModuleKindESNext = core.ModuleKindESNext
const ModuleKindNode16 = core.ModuleKindNode16
const ModuleKindNode18 = core.ModuleKindNode18
const ModuleKindNodeNext = core.ModuleKindNodeNext
const ModuleKindNone = core.ModuleKindNone
const ModuleKindPreserve = core.ModuleKindPreserve
const ModuleKindSystem = core.ModuleKindSystem
var ModuleKindToModuleResolutionKind = core.ModuleKindToModuleResolutionKind
const ModuleKindUMD = core.ModuleKindUMD
type ModuleResolutionKind = core.ModuleResolutionKind
const ModuleResolutionKindBundler = core.ModuleResolutionKindBundler
const ModuleResolutionKindNode16 = core.ModuleResolutionKindNode16
const ModuleResolutionKindNodeNext = core.ModuleResolutionKindNodeNext
const ModuleResolutionKindUnknown = core.ModuleResolutionKindUnknown
type NewLineKind = core.NewLineKind
const NewLineKindCRLF = core.NewLineKindCRLF
const NewLineKindLF = core.NewLineKindLF
const NewLineKindNone = core.NewLineKindNone
//go:linkname NewTextRange github.com/microsoft/typescript-go/internal/core.NewTextRange
func NewTextRange(pos int, end int) core.TextRange
//go:linkname NewWorkGroup github.com/microsoft/typescript-go/internal/core.NewWorkGroup
func NewWorkGroup(singleThreaded bool) core.WorkGroup
//go:linkname NonRelativeModuleNameForTypingCache github.com/microsoft/typescript-go/internal/core.NonRelativeModuleNameForTypingCache
func NonRelativeModuleNameForTypingCache(moduleName string) string
type ParsedOptions = core.ParsedOptions
type Pattern = core.Pattern
type PollingKind = core.PollingKind
const PollingKindDynamicPriority = core.PollingKindDynamicPriority
const PollingKindFixedChunkSize = core.PollingKindFixedChunkSize
const PollingKindFixedInterval = core.PollingKindFixedInterval
const PollingKindNone = core.PollingKindNone
const PollingKindPriorityInterval = core.PollingKindPriorityInterval
type Pool[T any] = core.Pool[T]
//go:linkname PositionToLineAndCharacter github.com/microsoft/typescript-go/internal/core.PositionToLineAndCharacter
func PositionToLineAndCharacter(position int, lineStarts []core.TextPos) (line int, character int)
type ProjectReference = core.ProjectReference
type ResolutionMode = core.ResolutionMode
const ResolutionModeCommonJS = core.ResolutionModeCommonJS
const ResolutionModeESM = core.ResolutionModeESM
const ResolutionModeNone = core.ResolutionModeNone
//go:linkname ResolveProjectReferencePath github.com/microsoft/typescript-go/internal/core.ResolveProjectReferencePath
func ResolveProjectReferencePath(ref *core.ProjectReference) string
type ScriptKind = core.ScriptKind
const ScriptKindDeferred = core.ScriptKindDeferred
const ScriptKindExternal = core.ScriptKindExternal
const ScriptKindJS = core.ScriptKindJS
const ScriptKindJSON = core.ScriptKindJSON
const ScriptKindJSX = core.ScriptKindJSX
const ScriptKindTS = core.ScriptKindTS
const ScriptKindTSX = core.ScriptKindTSX
const ScriptKindUnknown = core.ScriptKindUnknown
type ScriptTarget = core.ScriptTarget
const ScriptTargetES2015 = core.ScriptTargetES2015
const ScriptTargetES2016 = core.ScriptTargetES2016
const ScriptTargetES2017 = core.ScriptTargetES2017
const ScriptTargetES2018 = core.ScriptTargetES2018
const ScriptTargetES2019 = core.ScriptTargetES2019
const ScriptTargetES2020 = core.ScriptTargetES2020
const ScriptTargetES2021 = core.ScriptTargetES2021
const ScriptTargetES2022 = core.ScriptTargetES2022
const ScriptTargetES2023 = core.ScriptTargetES2023
const ScriptTargetES2024 = core.ScriptTargetES2024
const ScriptTargetES3 = core.ScriptTargetES3
const ScriptTargetES5 = core.ScriptTargetES5
const ScriptTargetESNext = core.ScriptTargetESNext
const ScriptTargetJSON = core.ScriptTargetJSON
const ScriptTargetLatest = core.ScriptTargetLatest
const ScriptTargetNone = core.ScriptTargetNone
//go:linkname ShouldRewriteModuleSpecifier github.com/microsoft/typescript-go/internal/core.ShouldRewriteModuleSpecifier
func ShouldRewriteModuleSpecifier(specifier string, compilerOptions *core.CompilerOptions) bool
type SourceFileAffectingCompilerOptions = core.SourceFileAffectingCompilerOptions
type Stack[T any] = core.Stack[T]
//go:linkname StringifyJson github.com/microsoft/typescript-go/internal/core.StringifyJson
func StringifyJson(input any, prefix string, indent string) (string, error)
const TSFalse = core.TSFalse
const TSTrue = core.TSTrue
const TSUnknown = core.TSUnknown
type TextChange = core.TextChange
type TextPos = core.TextPos
type TextRange = core.TextRange
type Tristate = core.Tristate
//go:linkname TryParsePattern github.com/microsoft/typescript-go/internal/core.TryParsePattern
func TryParsePattern(pattern string) core.Pattern
type TypeAcquisition = core.TypeAcquisition
//go:linkname UndefinedTextRange github.com/microsoft/typescript-go/internal/core.UndefinedTextRange
func UndefinedTextRange() core.TextRange
var UnprefixedNodeCoreModules = core.UnprefixedNodeCoreModules
//go:linkname Version github.com/microsoft/typescript-go/internal/core.Version
func Version() string
//go:linkname VersionMajorMinor github.com/microsoft/typescript-go/internal/core.VersionMajorMinor
func VersionMajorMinor() string
type WatchDirectoryKind = core.WatchDirectoryKind
const WatchDirectoryKindDynamicPriorityPolling = core.WatchDirectoryKindDynamicPriorityPolling
const WatchDirectoryKindFixedChunkSizePolling = core.WatchDirectoryKindFixedChunkSizePolling
const WatchDirectoryKindFixedPollingInterval = core.WatchDirectoryKindFixedPollingInterval
const WatchDirectoryKindNone = core.WatchDirectoryKindNone
const WatchDirectoryKindUseFsEvents = core.WatchDirectoryKindUseFsEvents
type WatchFileKind = core.WatchFileKind
const WatchFileKindDynamicPriorityPolling = core.WatchFileKindDynamicPriorityPolling
const WatchFileKindFixedChunkSizePolling = core.WatchFileKindFixedChunkSizePolling
const WatchFileKindFixedPollingInterval = core.WatchFileKindFixedPollingInterval
const WatchFileKindNone = core.WatchFileKindNone
const WatchFileKindPriorityPollingInterval = core.WatchFileKindPriorityPollingInterval
const WatchFileKindUseFsEvents = core.WatchFileKindUseFsEvents
const WatchFileKindUseFsEventsOnParentDirectory = core.WatchFileKindUseFsEventsOnParentDirectory
type WatchOptions = core.WatchOptions
//go:linkname WithLocale github.com/microsoft/typescript-go/internal/core.WithLocale
func WithLocale(ctx context.Context, locale language.Tag) context.Context
//go:linkname WithRequestID github.com/microsoft/typescript-go/internal/core.WithRequestID
func WithRequestID(ctx context.Context, id string) context.Context
type WorkGroup = core.WorkGroup
