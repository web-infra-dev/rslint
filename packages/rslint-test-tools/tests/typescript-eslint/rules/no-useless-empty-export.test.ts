import { describe, test, expect } from '@rstest/core';
import { noFormat, RuleTester } from '@typescript-eslint/rule-tester';
import { getFixturesRootDir } from '../RuleTester.ts';

const rootPath = getFixturesRootDir();
// this rule tests the spacing, which prettier will want to fix and break the tests
/* eslint "@typescript-eslint/internal/plugin-test-formatting": ["error", { formatWithPrettier: false }] */

const ruleTester = new RuleTester({
  languageOptions: {
    parserOptions: {
      project: './tsconfig.json',
      tsconfigRootDir: rootPath,
    },
  },
});

const error = {
  messageId: 'uselessExport',
} as const;

describe('no-useless-empty-export', () => {
  test('should work', () => {
    ruleTester.run('no-useless-empty-export', {
      valid: [
        "declare module '_'",
        "import {} from '_';",
        "import * as _ from '_';",
        'export = {};',
        'export = 3;',
        'export const _ = {};',
        `
      const _ = {};
      export default _;
    `,
        `
      export * from '_';
      export = {};
    `,
        `
      export {};
    `,
        // https://github.com/microsoft/TypeScript/issues/38592
        {
          code: `
        export type A = 1;
        export {};
      `,
          // @ts-ignore
          filename: 'foo.d.ts',
        },
        {
          code: `
        export declare const a = 2;
        export {};
      `,
          // @ts-ignore
          filename: 'foo.d.ts',
        },
        {
          code: `
        import type { A } from '_';
        export {};
      `,
          // @ts-ignore
          filename: 'foo.d.ts',
        },
        {
          code: `
        import { A } from '_';
        export {};
      `,
          // @ts-ignore
          filename: 'foo.d.ts',
        },
      ],
      invalid: [
        {
          code: `
export const _ = {};
export {};
      `,
          errors: [error],
          output: `
export const _ = {};

      `,
        },
        {
          code: `
export * from '_';
export {};
      `,
          errors: [error],
          output: `
export * from '_';

      `,
        },
        {
          code: `
export {};
export * from '_';
      `,
          errors: [error],
          output: `

export * from '_';
      `,
        },
        {
          code: `
const _ = {};
export default _;
export {};
      `,
          errors: [error],
          output: `
const _ = {};
export default _;

      `,
        },
        {
          code: `
export {};
const _ = {};
export default _;
      `,
          errors: [error],
          output: `

const _ = {};
export default _;
      `,
        },
        {
          code: `
const _ = {};
export { _ };
export {};
      `,
          errors: [error],
          output: `
const _ = {};
export { _ };

      `,
        },
        {
          code: `
import _ = require('_');
export {};
      `,
          errors: [error],
          output: `
import _ = require('_');

      `,
        },
        {
          code: `
import _ = require('_');
export {};
export {};
      `,
          errors: [error, error],
          output: `
import _ = require('_');


      `,
        },
      ],
    });
  });
});
