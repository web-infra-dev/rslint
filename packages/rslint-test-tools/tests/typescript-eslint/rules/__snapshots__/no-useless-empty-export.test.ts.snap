// Rstest Snapshot v1

exports[`no-useless-empty-export > invalid 1`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "Empty export does nothing and can be removed.",
      "messageId": "uselessExport",
      "range": {
        "end": {
          "column": 11,
          "line": 3,
        },
        "start": {
          "column": 1,
          "line": 3,
        },
      },
      "ruleName": "no-useless-empty-export",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`no-useless-empty-export > invalid 2`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "Empty export does nothing and can be removed.",
      "messageId": "uselessExport",
      "range": {
        "end": {
          "column": 11,
          "line": 3,
        },
        "start": {
          "column": 1,
          "line": 3,
        },
      },
      "ruleName": "no-useless-empty-export",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`no-useless-empty-export > invalid 3`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "Empty export does nothing and can be removed.",
      "messageId": "uselessExport",
      "range": {
        "end": {
          "column": 11,
          "line": 2,
        },
        "start": {
          "column": 1,
          "line": 2,
        },
      },
      "ruleName": "no-useless-empty-export",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`no-useless-empty-export > invalid 4`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "Empty export does nothing and can be removed.",
      "messageId": "uselessExport",
      "range": {
        "end": {
          "column": 11,
          "line": 4,
        },
        "start": {
          "column": 1,
          "line": 4,
        },
      },
      "ruleName": "no-useless-empty-export",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`no-useless-empty-export > invalid 5`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "Empty export does nothing and can be removed.",
      "messageId": "uselessExport",
      "range": {
        "end": {
          "column": 11,
          "line": 2,
        },
        "start": {
          "column": 1,
          "line": 2,
        },
      },
      "ruleName": "no-useless-empty-export",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`no-useless-empty-export > invalid 6`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "Empty export does nothing and can be removed.",
      "messageId": "uselessExport",
      "range": {
        "end": {
          "column": 11,
          "line": 4,
        },
        "start": {
          "column": 1,
          "line": 4,
        },
      },
      "ruleName": "no-useless-empty-export",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`no-useless-empty-export > invalid 7`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "Empty export does nothing and can be removed.",
      "messageId": "uselessExport",
      "range": {
        "end": {
          "column": 11,
          "line": 3,
        },
        "start": {
          "column": 1,
          "line": 3,
        },
      },
      "ruleName": "no-useless-empty-export",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`no-useless-empty-export > invalid 8`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "Empty export does nothing and can be removed.",
      "messageId": "uselessExport",
      "range": {
        "end": {
          "column": 11,
          "line": 3,
        },
        "start": {
          "column": 1,
          "line": 3,
        },
      },
      "ruleName": "no-useless-empty-export",
    },
    {
      "filePath": "src/virtual.ts",
      "message": "Empty export does nothing and can be removed.",
      "messageId": "uselessExport",
      "range": {
        "end": {
          "column": 11,
          "line": 4,
        },
        "start": {
          "column": 1,
          "line": 4,
        },
      },
      "ruleName": "no-useless-empty-export",
    },
  ],
  "errorCount": 2,
  "fileCount": 1,
  "ruleCount": 1,
}
`;
